lib/Core/TableStructure.js

```diff
@@ -182,13 +182,17 @@ TableStructure.fromCsv = function(csvString, result) {
     var json = csv.toArrays(csvString, {
         onParseValue: castToScalar
     });
-    // Handle a double line feed at the end: could improve this to remove more than one.
-    if (json.length > 1) {
-        var last = json[json.length - 1];
-        if (last.length === 1 && last[0] === null) {
-            json.pop();
+    // Remove extra blank lines at the end.
+    function removeFinalBlank(json) {
+        if (json.length > 1) {
+            var last = json[json.length - 1];
+            if (last.length === 1 && last[0] === null) {
+                json.pop();
+                removeFinalBlank(json);
```

Comment:
JavaScript doesn't support tail recursion (I'm pretty sure), so this will blow out the stack if there are a lot of blank lines at the end.  Why not just use a loop and a single call to splice?